grass/kernel.c

```diff
@@ -87,6 +87,10 @@ static void proc_yield() {
     /* Set the current process status to RUNNABLE if it was RUNNING */
     if (!CORE_IDLE && curr_status == PROC_RUNNING) proc_set_runnable(curr_pid);
 
+  /* Student's code goes here (preemptive scheduler)
+   * Capture current process's info before it is scheduled out.
```

Comment:
Say "Measure and record scheduling metrics for the current process before it yields."